File: src/lib/types/portfolio.ts
export interface PortfolioDashboardPointer {
	id: string;
	portfolio_dashboard_id: string;
	repo_dashboard_id: string;
	sort_order: number;
}

export interface AddRepoToPortfolioRequest {
	portfolio_dashboard_id: string;
	repo_dashboard_id: string;
}

File: src-tauri/src/models/portfolio.rs
use serde::{Deserialize, Serialize};
use ts_rs::TS;

#[derive(Debug, Clone, Serialize, Deserialize, TS)]
#[ts(export)]
pub struct PortfolioDashboardPointer {
    pub id: String,
    pub portfolio_dashboard_id: String,
    pub repo_dashboard_id: String,
    #[ts(type = "number")]
    pub sort_order: i64,
}

#[derive(Debug, Deserialize)]
pub struct AddRepoToPortfolioRequest {
    pub portfolio_dashboard_id: String,
    pub repo_dashboard_id: String,
}
